### PR TITLE
[web][api][SIMS #1300]Education Programs API Reorganization

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- Education program API reorganization.
- Removed similar methods.
- No actual changes in existing features.
- Changes affect students, AEST, and institutions.

### CODE REVIEW

File: sources/packages/api/src/route-controllers/education-program/education-program.students.controller.ts [hunk 1]
```diff
@@ -45,4 +54,36 @@ export class EducationProgramStudentsController extends BaseController {
...
+   * program years are considered.
+   * @returns key/value pair list of programs.
+   */
+  @Get("location/:locationId/program-year/:programYearId/options-list")
+  async getLocationProgramsOptionList(
+    @Param("locationId", ParseIntPipe) locationId: number,
+    @Param("programYearId", ParseIntPipe) programYearId: number,
+    @Query(
+      "isIncludeInActiveProgramYear",
+      new DefaultValuePipe(false),
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/education-program/education-program.students.controller.ts:73]: 👍

File: sources/packages/web/src/services/http/EducationProgramApi.ts [hunk 2]
```diff
@@ -1,118 +1,100 @@
...
+   * Get the programs summary of an institution with pagination.
+   * @param institutionId id of the institution.
+   * @param paginationOptions pagination options.
+   * @returns paginated programs summary.
+   */
+  async getProgramsSummaryByInstitutionId(
+    institutionId: number,
+    paginationOptions: PaginationOptions,
+  ): Promise<PaginatedResultsAPIOutDTO<EducationProgramsSummaryAPIOutDTO>> {
+    console.log(paginationOptions);
```
Comment (ann-aot) [inline sources/packages/web/src/services/http/EducationProgramApi.ts:44]: pls remove the console.log

File: sources/packages/web/src/services/http/EducationProgramApi.ts [hunk 3]
```diff
@@ -1,118 +1,100 @@
...
+    const url =
+      `education-program/institution/${institutionId}/summary?` +
```
Comment (ann-aot) [inline sources/packages/web/src/services/http/EducationProgramApi.ts:46]: we can also do, 
image

File: sources/packages/api/src/route-controllers/education-program/education-program.controller.service.ts [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+      entranceRequirements: {
+        hasMinimumAge: program.hasMinimumAge,
+        minHighSchool: program.minHighSchool,
+        requirementsByInstitution: program.requirementsByInstitution,
+        requirementsByBCITA: program.requirementsByBCITA,
+      },
+      hasWILComponent: program.hasWILComponent,
+      isWILApproved: program.isWILApproved,
+      wilProgramEligibility: program.wilProgramEligibility,
+      hasTravel: program.hasTravel,
```
Comment (henryoforeh-aot) [inline sources/packages/api/src/route-controllers/education-program/education-program.controller.service.ts:188]: Please what does hasTravel mean ? Is this supposed to be like travel allowance ?

File: sources/packages/web/src/services/http/EducationProgramApi.ts [hunk 5]
```diff
@@ -1,118 +1,100 @@
...
-   * DEFAULT_PAGE_NUMBER is taken
-   * @param pageLimit, limit of the page if nothing is
-   * passed then DEFAULT_PAGE_LIMIT is taken
-   * @param searchCriteria, program name keyword to be searched
-   * @param sortField, field to be sorted
-   * @param sortOrder, order to be sorted
-   * @returns program summary for an institution location.
+   * Creates a new education program.
+   * @param payload information to create the new program.
+   * @returns id of the created program.
```
Comment (ann-aot) [inline sources/packages/web/src/services/http/EducationProgramApi.ts:69]: no need of return

File: sources/packages/web/src/types/contracts/EducationProgram.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+export interface ProgramOfferingHeader {
+  institutionId: number;
+  institutionName: string;
+  submittedDate: string;
+  status: ProgramStatus | OfferingStatus;
+  assessedBy: string;
+  assessedDate: string;
+  effectiveEndDate?: Date; //this field is only for programs.
```
Comment (ann-aot) [inline sources/packages/web/src/types/contracts/EducationProgram.ts:52]: uppercase `T`. Same for line 53

**Review by ann-aot — COMMENTED** [review]
LGTM 👍 some minor comments

**Review by henryoforeh-aot — APPROVED** [review]
Please take a look at the minor comment I wrote.

File: sources/packages/web/src/views/institution/locations/programs/LocationPrograms.vue [hunk 7]
```diff
@@ -136,13 +136,15 @@ export default {
     ) => {
       loading.value = true;
       programAndCount.value =
-        await EducationProgramService.shared.getLocationProgramsSummary(
+        await EducationProgramService.shared.getProgramsSummaryByLocationId(
```
Comment (guru-aot) [inline sources/packages/web/src/views/institution/locations/programs/LocationPrograms.vue:139]: 👍

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrewsignori-aot

File: sources/packages/api/src/route-controllers/education-program/education-program.students.controller.ts (hunk 1 above)
Comment (dheepak-aot) [inline sources/packages/api/src/route-controllers/education-program/education-program.students.controller.ts:73]: 👍

File: sources/packages/web/src/types/contracts/EducationProgram.ts [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+import { OfferingStatus } from "@/types";
+
+/**
+ * Program intensities of the education program.
+ */
+export enum ProgramIntensity {
+  /**
+   * Program with ProgramIntensity = Full Time and Part Time, will be both Full Time and Part Time
+   */
+  fullTimePartTime = "Full Time and Part Time",
```
Comment (dheepak-aot) [inline sources/packages/web/src/types/contracts/EducationProgram.ts:11]: Not a part of the PR, can we change the enum to start with upper case? there were only 3 occurrences in the code.
Also fullTimePartTime can be changed to FullTimeAndPartTime. Just a suggestion.

**Review by dheepak-aot — APPROVED** [review]
Great work @andrewsignori-aot. Refactoring program is very challenging as it has a lot of code written long back. 👍
One minor comment.

File: sources/packages/api/src/route-controllers/education-program/education-program.controller.service.ts (hunk 4 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/education-program/education-program.controller.service.ts:188]: I do not have much information about it. The comments say:
"Identifier for Field trip, field placement, or travel component in program."
It was introduced 8 months ago.

File: sources/packages/web/src/types/contracts/EducationProgram.ts (hunk 8 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/types/contracts/EducationProgram.ts:11]: About the upper case, we need to do it globally.
About the `FullTimeAndPartTime`, the way that it is is the way that it should be displayed to the users.

**Review by ann-aot — APPROVED** [review]
Thanks for doing the changes 👍

**Pull request merged**